### ISSUE TITLE
(bedrock): add inference profiles / cross-region inference

### ISSUE THREAD

**Issue by aws-rafams** [awslabs/generative-ai-cdk-constructs#683]
### Describe the feature

An inference profile in the context of Amazon Bedrock is a configuration that allows you to route model inference traffic to multiple AWS Regions for increased availability and throughput.
https://docs.aws.amazon.com/bedrock/latest/userguide/cross-region-inference-support.html
![Screenshot 2024-09-04 at 11 11 30](https://github.com/user-attachments/assets/ff5c15f6-8a6b-485c-8782-c46b47a41379)

### Use Case

Add inference profiles for improved availability and to access certain models which are only available in certain regions through the usage of inference profiles. ![Screenshot 2024-09-04 at 11 16 19](https://github.com/user-attachments/assets/9ceabefd-e867-406c-8676-d246ca31cfcb)

### Proposed Solution

Define `BedrockInferenceProfile` as follows:
```ts
export enum InferenceProfileRegion {
  /**
   * EU: Frankfurt (eu-central-1), Ireland (eu-west-1), Paris (eu-west-3)
   */
  EU = 'eu',
  /**
   * US: N. Virginia (us-east-1), Oregon (us-west-2)
   */
  US = 'us',
}

export interface InferenceProfileProps {
  /**
   * The geo region where the traffic is going to be distributed.
   */
  readonly region: InferenceProfileRegion;
  /**
   * A model supporting cross-region inference.
   * @see https://docs.aws.amazon.com/bedrock/latest/userguide/cross-region-inference-support.html
   */
  readonly model: BedrockFoundationModel;
}

export class BedrockInferenceProfile extends Resource {
  /**
   * @example 'us.anthropic.claude-3-5-sonnet-20240620-v1:0'
   */
  public readonly profileId: string;
  /**
   * @example 'arn:aws:bedrock:us-east-1:123456789012:inference-profile/us.anthropic.claude-3-5-sonnet-20240620-v1:0'
   */
  public readonly profileArn: string;
  constructor(scope: IConstruct, id: string, props: InferenceProfileProps) {
    super(scope, id);
    this.profileId = `${props.geoRegion}.${props.model.modelId}`;
    this.profileArn = Arn.format({
      service: 'bedrock',
      resource: 'inference-profile',
      resourceName: this.profileId,
      arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
    }, Stack.of(scope));
  }
}
```

doing so, the definition of the Inference Profile would be as follows:
```ts
new BedrockInferenceProfile(this, 'InferenceProfile', {
  region: InferenceProfileRegion.EU,
  model: BedrockFoundationModel.ANTHROPIC_CLAUDE_SONNET_V1_0,
})
```

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by aws-rafams** [comment]
Support in Knowledge Bases and thus Prompt Management [has arrived](https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-bedrock-knowledge-bases-cross-region-inference). Will create a PR shortly.

**Comment by aws-rafams** [comment]
CFN Spec [updated](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-bedrock-applicationinferenceprofile.html), awaiting release in CDK.

**Comment by krokoko** [comment]
It is available as of [CDK v2.166](https://github.com/aws/aws-cdk/releases/tag/v2.166.0) -> https://github.com/aws/aws-cdk/pull/32007

**Comment by krokoko** [comment]
Referencing https://github.com/awslabs/generative-ai-cdk-constructs/issues/796
